src/QMCDrivers/DMC/DMCUpdatePbyPFast.cpp

```diff
@@ -83,13 +83,11 @@ void DMCUpdatePbyPWithRejectionFast::advanceWalker(Walker_t& thisWalker, bool re
       dr += sqrttau * deltaR[iat];
       RealType rr = tauovermass * dot(deltaR[iat], deltaR[iat]);
       rr_proposed += rr;
-      if (rr > m_r2max)
+      if (!W.makeMoveAndCheck(iat, dr) || rr > m_r2max)
```

Comment:
No function calls like this in if statements please. The execution order is not obvious. Split.

Reply:
Rationale: many people will not know if makeMoveAndCheck is always called (or not).

Reply:
Actually to me
1. makeMoveAndCheck is solid.
2. rr > m_r2max is a hack. Not clear what is this actually for. This type of hack always sounds like to make bad WF caused bad moves to continue.

Reply:
Getting rid of the maximum displacement check is a discussion for another day.

Interesting that the intent to do something smart about large moves -- which should not happen in healthy runs -- might have precipitated a far more serious bug that affected actually healthy runs.

Reply:
Just pull the call() out of the if statements for now.

Reply:
Done. All pulled out.